presto-main/src/main/java/com/facebook/presto/execution/SqlTask.java

```diff
@@ -78,6 +83,8 @@
     private final AtomicReference<TaskHolder> taskHolderReference = new AtomicReference<>(new TaskHolder());
     private final AtomicBoolean needsPlan = new AtomicBoolean(true);
 
+    private final AtomicReference<Optional<TaskClientOutputContext>> clientOutputContext = new AtomicReference<>(Optional.empty());
```

Comment:
Getter is not synchronized, so the field needs to be marked as "volatile" for an update to be seen by another thread. Throughout Presto codebase I see atomic primitives are commonly used in such cases instead of volatile.